src/main/java/no/unit/nva/cristin/institutions/model/Identifier.java

```diff
@@ -0,0 +1,154 @@
+package no.unit.nva.cristin.institutions.model;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+public class Identifier {
+    public static final String IDENTIFIER_TEMPLATE = "%s.%s.%s.%s";
+    public static final String EMPTY_SUBUNIT = "0";
+    public static final String MATCH_SEPARATOR = "\\.";
+    public static final int FOUR_SEGMENTS = 4;
+
+    private String institution;
+    private String faculty;
+    private String department;
+    private String section;
+
+    public String getInstitution() {
+        return institution;
+    }
+
+    public void setInstitution(String institution) {
+        this.institution = institution;
+    }
+
+    public String getFaculty() {
+        return faculty;
+    }
+
+    public void setFaculty(String faculty) {
+        this.faculty = faculty;
+    }
+
+    public String getDepartment() {
+        return department;
+    }
+
+    public void setDepartment(String department) {
+        this.department = department;
+    }
+
+    public String getSection() {
+        return section;
+    }
+
+    public void setSection(String section) {
+        this.section = section;
+    }
+
+
+    /**
+     * An object to hold identifiers composed according to pattern "{institution}.{faculty}.{department}.{section}".
+     *
+     * @param identifier A dot-separated string formatted according to pattern {integer}.{integer}.{integer}.{integer}
+     * @throws BadlyFormattedIdentifierException In case the identifier does not match the input pattern
+     */
+    public Identifier(String identifier) {
+        if (isNull(identifier) || identifier.isEmpty()) {
+            throw new BadlyFormattedIdentifierException(identifier);
+        }
+        if (identifier.split(MATCH_SEPARATOR).length == FOUR_SEGMENTS) {
+            String[] segments = identifier.split(MATCH_SEPARATOR);
+            this.institution  = segments[0];
+            this.faculty = segments[1];
+            this.department = segments[2];
+            this.section = segments[3];
+        } else {
+            throw new BadlyFormattedIdentifierException(identifier);
+        }
+    }
+
+    /**
+     * A method to create a top-level unit.
+     *
+     * @param institution An integer representing the top-level unit
+     */
+    public Identifier(int institution) {
+        this(String.format(IDENTIFIER_TEMPLATE,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: institution (declared at line 76). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 77. Consider formatting this data with &#39;%d&#39; instead.